webapi/CopilotChat/Models/BotResponsePrompt.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System.Text.Json.Serialization;
+
+namespace SemanticKernel.Service.CopilotChat.Models;
+
+/// <summary>
+/// The fianl prompt sent to generate bot response.
+/// </summary>
+public class BotResponsePrompt
+{
+    /// <summary>
+    /// The system persona of the chat, includes SystemDescription and SystemResponse components from PromptsOptions.cs.
+    /// </summary>
+    [JsonPropertyName("systemPersona")]
+    public string SystemPersona { get; set; } = string.Empty;
+
+    /// <summary>
+    /// Audience extracted from conversation history.
+    /// </summary>
+    [JsonPropertyName("audience")]
+    public string Audience { get; set; } = string.Empty;
+
+    /// <summary>
+    /// User intent extracted from input and conversation history.
+    /// </summary>
+    [JsonPropertyName("userIntent")]
+    public string UserIntent { get; set; } = string.Empty;
+
+    /// <summary>
+    /// Chat memories queried from the chat memory store if any, includes long term and working memory.
+    /// </summary>
+    [JsonPropertyName("chatMemories")]
+    public string PastMemories { get; set; } = string.Empty;
+
+    /// <summary>
+    /// Relevant additional knowledge extracted using a planner.
+    /// </summary>
+    [JsonPropertyName("externalInformation")]
+    public string ExternalInformation { get; set; } = string.Empty;
+
+    /// <summary>
+    /// Preamble to the LLM's response.
+    /// </summary>
+    [JsonPropertyName("systemChatContinuation")]
+    public string SystemChatContinuation { get; set; } = string.Empty;
+
+    /// <summary>
+    /// Raw content of the prompt. Used to pass rendered prompt around ChatSkill.
+    /// </summary>
+    [JsonIgnore]
+    public string RawContent { get; set; } = string.Empty;
+
+    public BotResponsePrompt(
+        string rawContent,
+        string systemDescription,
+        string systemResponse,
+        string audience,
+        string userIntent,
+        string chatMemories,
+        string documentMemories,
+        string externalInformation,
+        string systemChatContinuation
+        )
+    {
+        this.RawContent = rawContent;
+        this.SystemPersona = string.Join("\n", systemDescription, systemResponse);
+        this.Audience = audience;
+        this.UserIntent = userIntent;
+        this.PastMemories =
+            !string.IsNullOrEmpty(chatMemories) && !string.IsNullOrEmpty(chatMemories)
+            ? string.Join("\n", chatMemories, documentMemories)
+            : chatMemories ?? documentMemories;
```

Comment:
Fixed the chatmemories issues
Do you have a suggestion on how to improve it?

Reply:
I changed it to this. Lmk what you think
```
...
this.PastMemories = chatMemories ?? documentMemories;
        if (!string.IsNullOrEmpty(chatMemories) && !string.IsNullOrEmpty(documentMemories))
        {
            this.PastMemories = string.Join("\n", chatMemories, documentMemories);
        }
...
```

Reply:
I think my confusion mainly comes from the fact that chatMemories is validated twice. My suggestion would be
```
this.PastMemories = chatMemories ?? '';
if (!string.IsNullOrEmpty(documentMemories))
{
    this.PastMemories = string.Join("\n", this.PastMemories, documentMemories).Trim();
}
```

Reply:
The issue is that chatMemories can be an empty string (I don't think it's ever just null), so chatMemories ?? '' would always be at least an empty string.

Also why string.Join doesn't just return an empty string when there's no memories because chatMemories and documetnMemories are empty strings when there are no memories, so they count as objects. So when there are no relevant memories, it returns just "\n"

Reply:
Can PastMemories be null?

Reply:
changed to 
```
        this.PastMemories = string.Join("\n", chatMemories, documentMemories).Trim();
```

Reply:
![image](https://github.com/microsoft/chat-copilot/assets/125500434/a0d12008-608a-4f9c-b444-65a236c45095)